### ISSUE TITLE
Can I run an ECO flow even if there aren't any timing violations?

### ISSUE THREAD

**Issue by jarvis-lorry** [The-OpenROAD-Project/OpenLane#1218]
Is it possible run the complete OL flow, change something in the LEF or DEF files and then have the tool rerun the route, keeping everything placed the same?  I don't have any timing violations, but I'm trying to see how changes to the library affect routing.  Does the current ECO flow allow for this?

**Comment by vijayank88** [comment]
Use interactive mode. So that you can use updated file as `CURRENT_DEF`

**Comment by donn** [comment]
@jarvis-lorry As @vijayank88 stated, you can use interactive mode (`./flow.tcl -it`) which allows you to run functions from the flow step-by-step, but you can also try `./flow.tcl -last_run -from routing` after your modifications, for example.

**Comment by jarvis-lorry** [comment]
Thank you, @vijayank88 and @donn.  @vijayank88, using the interactive flow does work for me but occasionally, OL errors out, probably from a mistake I've made.  @donn, I will try your suggestion today.

**Comment by jarvis-lorry** [comment]
> @jarvis-lorry As @vijayank88 stated, you can use interactive mode (`./flow.tcl -it`) which allows you to run functions from the flow step-by-step, but you can also try `./flow.tcl -last_run -from routing` after your modifications, for example.

@donn , if I use your second option, how do I set the CURRENT_LEF and MERGED_LEF_UNPADDED?

**Comment by donn** [comment]
`CURRENT_DEF`, you mean? There's a `config.tcl` file in the run. You can modify the last `CURRENT_DEF` in it, same for `MERGED_LEF_UNPADDED`.

You can also try passing `-override_env CURRENT_DEF=path_to_x,MERGED_LEF_UNPADDED=path_to_y` but I must confess I haven't tested that.

**Comment by jarvis-lorry** [comment]
@donn I've tried to use your suggestion but I'm having problems.  I can run './flow -design spm -last_run -from routing' but I cannot change the CURRENT_DEF because no change I make to config.tcl is reflected in the run.  In the run directory are two config files, config.tcl and config_in.tcl, and these are overwritten by OpenLane after each run.  Even if I change the value of CURRENT_DEF in config.tcl, it's ignored and overwritten.  OpenLane now uses config.json in the design directory when it starts compiling.  I've modified config.json but still I can't change the value of CURRENT_DEF.
